hibernate-redis/src/main/java/org/hibernate/cache/redis/jedis/JedisClient.java

```diff
@@ -432,6 +433,88 @@ public void execute(Transaction tx) {
     }
 
     /**
+     * We want nextTimestamp to return a long that is greater than previous calls to nextTimestamp.
+     *
+     * The simplest algorithm would be to ignore Redis and return currentTimeMillis. However, different VMs can have small drifts
```

Comment:
@johnou As you you know WATCH/MULTI is [deprecated stuff](http://redis.io/topics/transactions#redis-scripting-and-transactions) and lua works faster and should be used instead. Redisson uses lua in many places. And no WATCH/MULTI here is a [last issue](https://github.com/mrniko/redisson/commit/4fcd54ec8cc7457434d0eb50de52274658625f3c) when I switched from watch/multi to lua and gain 4x performance boost.

Here is example for your case:
```
redis.call('setnx', KEYS[1], ARGV[1]);
redis.call('incr', KEYS[1]);
```
KEYS[1] - key name
ARGV[1] - args value

Further information is here http://redis.io/commands/eval